tpie/serialization_sorter.h

```diff
@@ -886,8 +914,8 @@ class serialization_sorter {
 
 		memory_size_type finalFanoutMemory = m_params.memoryPhase3;
 		memory_size_type finalFanout =
-			std::min(fanout,
-					 finalFanoutMemory / perFanout);
+			std::min(std::min(fanout,
+					 finalFanoutMemory / perFanout), m_params.filesPhase3 - 1);
```

Comment:
why not use std::min({fanout, finalFanoutMemory / perFanout, m_params.filesPhase3 - 1})?